src/vos/evtree.c

```diff
@@ -969,6 +969,22 @@ evt_tcx_reset_trace(struct evt_context *tcx)
 	evt_tcx_set_trace(tcx, 0, tcx->tc_root->tr_node, 0, false);
 }
 
+static bool
+evt_tcx_reset(void *tcx)
+{
+	memset(tcx, 0, sizeof(struct evt_context));
+	return true;
+}
+
+struct d_dtm_reg vos_tcx_type = {
+		.dr_init = NULL,
+		.dr_reset = evt_tcx_reset,
+		.dr_release = NULL,
+		.dr_max_desc = 0,
+		.dr_max_free_desc = 0,
+		POOL_TYPE_INIT(evt_context, tc_dtm_link)
+};
+
```

Comment:
s/tcx/evt

We use tcx for open btree handles as well.   Since we only ever need one of these per ioc, I wonder if we shouldn't create an allocation routine and just keep one with the ioc that can be passed to open/create for reuse.   So something like the following

daos_handle_t evt_handle_alloc(void);
void evt_handle_free(daos_handle th);

then 
int evt_open/create_ex(...., th);  /* pass an allocated handle rather than relying on internal allocation */

This type of method would be more amenable to extending to dbtree